pkgs/by-name/en/en-croissant/package.nix

```diff
@@ -0,0 +1,124 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, stdenvNoCC
+, jq
+, moreutils
+, nodePackages
+, cacert
+, rustPlatform
+, cargo
+, rustc
+, cargo-tauri
+, pkg-config
+, openssl
+, libsoup
+, wrapGAppsHook
+, gtk3
+, webkitgtk
+}:
+
+stdenv.mkDerivation rec {
+  pname = "en-croissant";
+  version = "0.9.2";
+
+  src = fetchFromGitHub {
+    owner = "franciscoBSalgueiro";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-5Qt/OvsMw5G6Q9WN+OTdWVlJQlEz/F6C6z3zO05uOlM=";
+  };
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  pnpm-deps = stdenvNoCC.mkDerivation {
+    pname = "${pname}-pnpm-deps";
+    inherit src version;
+
+    nativeBuildInputs = [
+      jq
+      moreutils
+      nodePackages.pnpm
+      cacert
+    ];
+
+    pnpmPatch = builtins.toJSON {
+      pnpm.supportedArchitectures = {
+        os = [ "linux" ];
+        cpu = [ "x64" "arm64" ];
+      };
+    };
+
+    postPatch = ''
+      mv package.json package.json.orig
+      jq --raw-output ". * $pnpmPatch" package.json.orig > package.json
+    '';
+
+    installPhase = ''
```

Comment:
please keep the pre-/post-phase hooks
https://nix.dev/tutorials/packaging-existing-software#phases-and-hooks